core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -279,6 +280,7 @@ object KafkaConfig {
   val NewGroupCoordinatorEnableProp = "group.coordinator.new.enable"
   val GroupCoordinatorRebalanceProtocolsProp = "group.coordinator.rebalance.protocols"
   val GroupCoordinatorNumThreadsProp = "group.coordinator.threads"
+  val GroupProtocolMigrationPolicyProp = "group.protocol.migration.policy"
```

Comment:
nit: It may be better to have it scope as a "consumer" config. How about using `group.consumer.upgrade.policy`?